src/GraphQL/Validation/Rules/LoneAnonymousOperation.cs

```diff
@@ -17,20 +17,21 @@ public class LoneAnonymousOperation : IValidationRule
         /// </summary>
         public static readonly LoneAnonymousOperation Instance = new LoneAnonymousOperation();
 
-        /// <inheritdoc/>
-        /// <exception cref="LoneAnonymousOperationError"/>
-        public Task<INodeVisitor> ValidateAsync(ValidationContext context)
-        {
-            var operationCount = context.Document.Operations.Count;
-
-            return new MatchingNodeVisitor<Operation>(op =>
+        private static readonly Task<INodeVisitor> _task = new EnterLeaveListener(_ =>
+            {
+                _.Match<Document>((_, context) => context.Set<LoneAnonymousOperation>(context.Document.Operations.Count));
+                _.Match<Operation>((op, context) =>
                 {
-                    if (string.IsNullOrWhiteSpace(op.Name)
-                        && operationCount > 1)
+                    int operationCount = context.Get<LoneAnonymousOperation, int>();
```

Comment:
🤦 I could just use `context.Document.Operations.Count` directly.

Reply:
Take a look at what I did in my PR.  I short-circuited the entire operation.

Reply:
Already looking into. What do you mean _short-circuited_ (LoneAnonymousOperation.cs )?

Reply:
If you looked at the code, I assume you figured it out.  But basically, it just returns a "null" INodeVisitor after performing the context-wide check short-circuit check.

Reply:
I saw this in another file, not this one.